controllers/mars/pod.go

```diff
@@ -39,24 +35,10 @@ func (r *MarsJobReconciler) GetPodsForJob(obj interface{}) ([]*corev1.Pod, error
 	})
 	// List all pods to include those that don't match the selector anymore
 	// but have a ControllerRef pointing to this controller.
-	podlist := &corev1.PodList{}
-	err = r.List(context.Background(), podlist, client.MatchingLabelsSelector{Selector: selector})
+	podList := &corev1.PodList{}
```

Comment:
the entire method may be all be a common method too, anyway we can do it later, merge this in first